integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/withapim/soap/SOAPApisTestCase.java

```diff
@@ -0,0 +1,116 @@
+    void setEnvironment() throws Exception {
+        super.initWithSuperTenant();
+        // Get App ID and API ID
+        String applicationId = ApimResourceProcessor.applicationNameToId.get(APPLICATION_NAME);
+        String apiId = ApimResourceProcessor.apiNameToId.get(API_NAME);
+
+        String accessToken = StoreUtils.generateUserAccessToken(apimServiceURLHttps, applicationId,
+                user, storeRestClient);
+        requestHeaders_11.put(TestConstant.AUTHORIZATION_HEADER, "Bearer " + accessToken);
+        requestHeaders_11.put(TestConstant.CONTENT_TYPE_HEADER, TestConstant.CONTENT_TYPES.TEXT_XML);
+        requestHeaders_11.put(TestConstant.SOAP_ACTION_HEADER, SOAP_ACTION);
+        requestHeaders_12.put(TestConstant.AUTHORIZATION_HEADER, "Bearer " + accessToken);
+        requestHeaders_12.put(TestConstant.CONTENT_TYPE_HEADER, TestConstant.CONTENT_TYPES.SOAP_XML);
+        API api = new API();
+        api.setContext(API_CONTEXT + TestConstant.URL_SEPARATOR + "1.0.0");
+        api.setName(API_NAME);
+        api.setVersion("1.0.0");
+        api.setProvider("admin");
+
+        internalKey = TokenUtil.getJWT(api, null, "Unlimited", TestConstant.KEY_TYPE_PRODUCTION,
+                3600, null, true);
+        endpointURL11 = Utils.getServiceURLHttps(API_CONTEXT + TestConstant.URL_SEPARATOR +
+                API_VERSION + "/phoneverify11");
+        endpointURL12 = Utils.getServiceURLHttps(API_CONTEXT + TestConstant.URL_SEPARATOR +
+                API_VERSION + "/phoneverify12");
+    }
+
+    @Test(description = "Send a request to the subscribed SOAP API using SOAP 1.1")
+    public void testInvokeSoapAPIv11() throws CCTestException, InterruptedException {
+        HttpResponse response = HttpsClientRequest.doPost(endpointURL11,
+                TestConstant.SOAP_ENVELOPES.SOAP11_SAMPLE_REQ_PAYLOAD, requestHeaders_11);
+        Assert.assertNotNull(response, "Error occurred while invoking the endpoint " + endpointURL11 + ". HttpResponse");
+        Assert.assertEquals(response.getResponseCode(), HttpStatus.SC_SUCCESS,
+                "Valid subscription should be able to invoke the associated API");
+        Assert.assertTrue(response.getData().contains("<Valid>true</Valid>"), "Response body mismatched");
+        Map<String, String> responseHeaders = response.getHeaders().entrySet().stream().collect(
+                Collectors.toMap(
+                        entry -> entry.getKey().toLowerCase(),
+                        entry -> entry.getValue()
+                )
+        );
+        Assert.assertEquals(responseHeaders.get(TestConstant.CONTENT_TYPE_HEADER), TestConstant.CONTENT_TYPES.TEXT_XML,
+                "Response content-type mismatch");
+    }
+
+    @Test(description = "Send a request to the subscribed SOAP API using SOAP 1.2")
+    public void testInvokeSoapAPIv12() throws CCTestException, InterruptedException {
```

Comment:
Shall we add a test case for following?
1. Authentication Failure with soap1.1 and soap1.2 (to check fault message generation from enforcer)
2. No route found scenario with soap1.1 and soap1.2 (to check the response mapper behavior)

Reply:
fixed in https://github.com/wso2/product-microgateway/pull/2972/commits/02578c32134f02724d937c2ce88c6b64d8e0c505